--- netrexx-code/src/org/netrexx/process/RxTranslator.nrx.orig	Thu May  9 18:29:03 2019
+++ netrexx-code/src/org/netrexx/process/RxTranslator.nrx	Thu May  9 18:39:59 2019
@@ -44,6 +44,7 @@
 -- 2015.05.15 reformat build info  -- KK -- Kermit
 -- 2015.06.11 -- fix for #134 time output needs correct compiler name -- KK
 -- 2015.06.11 -- if using jsr199 to pass java code and no java output, don't touch disk files
+/* 2019.05.09 display failed summaries last - Jason Martin            */
 /* ------------------------------------------------------------------ */
 /* The translator originally was slightly more general, in that it    */
 /* permitted an output .java to be specified.  We now [1997.04.20]    */
@@ -868,6 +869,7 @@
     /* Arg1 is 1 if we had interpretation */
   method summary(hadexec=boolean)
     
+    bucket=Rexx ""
     /* Summaries, one per program */
     loop f=0 for programlist.length
       program=programlist[f]                        -- current program
@@ -902,11 +904,30 @@
           okresult='successful'
         end
 	
-	if ok then say(action 'of '''inname'''' okresult||add)
-        else say(action 'of '''inname''' failed'add)
+	--if ok then say(action 'of '''inname'''' okresult||add)
+        --else say(action 'of '''inname''' failed'add)
+        if ok then bucket[f]=action||" of \'"inname"\'"||okresult||add
+        else bucket[f]=action||" of \'"inname"\' failed"||add
       end summary
       
     end f
+    faulty=Rexx ""
+    x=0
+    loop skim=0 for programlist.length
+      if bucket[skim].lastpos("successful")\=0 then do
+        say bucket[skim]
+      end
+      else do
+        x=x+1
+        faulty[x]=bucket[skim]
+        faulty[0]=x
+      end
+    end
+    if faulty.exists("1") then do
+      loop review=1 to faulty[0]
+        say faulty[review]
+      end
+    end
     return
     
     
